flow/SystemMonitor.cpp

```diff
@@ -149,16 +149,15 @@ SystemStatistics customSystemMonitor(std::string eventName, StatisticsState *sta
 
 			for (int i = 0; i < NetworkMetrics::PRIORITY_BINS && g_network->networkMetrics.priorityBins[i] != TaskPriority::Zero; i++) {
 				if(g_network->networkMetrics.priorityBlocked[i]) {
-					double lastSegment = std::min(currentStats.elapsed, now() - g_network->networkMetrics.priorityTimer[i]);
-					g_network->networkMetrics.priorityBlockedDuration[i] += lastSegment;
-					g_network->networkMetrics.secSquaredPriorityBlocked[i] += lastSegment * lastSegment;
-					g_network->networkMetrics.priorityTimer[i] = now();
+					g_network->networkMetrics.priorityBlockedDuration[i] += std::min(currentStats.elapsed, now() - g_network->networkMetrics.windowedPriorityTimer[i]);
+					g_network->networkMetrics.priorityMaxBlockedDuration[i] = std::max(g_network->networkMetrics.priorityMaxBlockedDuration[i], now() - g_network->networkMetrics.priorityTimer[i]);
+					g_network->networkMetrics.windowedPriorityTimer[i] = now();
```

Comment:
because of the min on line 152, we do not need to reset windowedPriorityTimer, so we can probably get rid of it

Reply:
We do use `windowedPriorityTimer` in Net2 without a `min` if a busy period started in a previous measurement window and ends in the current one.

It looks like the reason for doing the `min` is because the way we measure elapsed is slightly different than how we measure priority busyness (using `timer()` instead of `now()`), so that helps to avoid going slightly above 100%. It may be better to remove the `min` from its current location, though, and instead put it on the line where we do the logging, as that would cover more cases.